### PR TITLE
fix: Remove double encoding for OG image URLs

### PR DESCRIPTION
**Pull request by hbjORbj**
## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] N/A - I have updated the developer docs in /docs if this PR makes changes that would require a [documentation change](https://cal.com/docs). If N/A, write N/A here and check the checkbox.
- [x] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

1. Login with `admin` account.
2. Go to some event type (e.g., http://localhost:3000/admin/quick-chat)
3. Find `meta og:image` element in `head` and run its URL in a new tab
4. See if it renders well

### CODE REVIEW

File: apps/web/pages/api/social/og/image.tsx [hunk 1]
```diff
@@ -87,7 +87,7 @@ export default async function handler(req: NextApiRequest) {
 
       return new Response(img.body, {
         status: 200,
-        headers: { "Content-Type": "image/png", "cache-control": "max-age=86400" },
```
Comment (hbjORbj) [inline apps/web/pages/api/social/og/image.tsx:90]: we initially thought this would fix the issue but it didn't

File: apps/web/next.config.js [hunk 2]
```diff
@@ -219,6 +219,7 @@ const nextConfig = {
     },
   },
   images: {
+    formats: ['image/avif', 'image/webp'],
```
Comment (hbjORbj) [inline apps/web/next.config.js:222]: PNG fallback happens automatically for browsers that don't support these formats

File: packages/lib/OgImages.tsx [hunk 3]
```diff
@@ -70,14 +67,14 @@ export const constructMeetingImage = (
     // Joining a multiline string for readability.
   ].join("");
 
-  return encodeUri ? encodeURIComponent(url) : url;
```
Comment (hbjORbj) [inline packages/lib/OgImages.tsx:73]: due to this double encoding, it results in 404, we already encode params here
Screenshot 2024-11-27 at 5 18 55 AM

File: apps/web/pages/api/social/og/image.tsx [hunk 4]
```diff
@@ -87,7 +87,7 @@ export default async function handler(req: NextApiRequest) {
...
+        headers: { "Content-Type": "image/png", "cache-control": "max-age=0" },
```
Comment (hbjORbj) [inline apps/web/pages/api/social/og/image.tsx:90]: Agreed with @keithwillcode. We will set this back to 86400 in the next release.

**Review by PeerRich — APPROVED** [review]

**Pull request merged**